test/topology/test_topology_failure_recovery.py

```diff
@@ -4,73 +4,105 @@
 # SPDX-License-Identifier: AGPL-3.0-or-later
 #
 from test.pylib.manager_client import ManagerClient
-from test.pylib.internal_types import ServerInfo
 from test.pylib.scylla_cluster import ReplaceConfig
 from test.topology.conftest import skip_mode
+from test.topology.util import get_topology_coordinator
 import pytest
 import logging
-import asyncio
+import random
+import sys
 
 logger = logging.getLogger(__name__)
 
-async def inject_error_on(manager, error_name, servers):
-    errs = [manager.api.enable_injection(s.ip_addr, error_name, True) for s in servers]
-    await asyncio.gather(*errs)
+
+def init_random_seed():
+    """ Initialize random seed for the test.
+
+        The seed is logged so that the test can be reproduced if needed.
+    """
+    seed = random.randrange(sys.maxsize)
+    random.seed(seed)
+    logger.info("Random seed: %s", seed)
+
+
+async def get_running_servers(manager: ManagerClient):
+    """ Return the running servers in randomized order.
+
+        This helps to avoid making any assumptions on the order of the servers,
+        as in general the order of the cluster running servers can be somewhat
+        random (especially when initialized in parallel).
+    """
+    servers = await manager.running_servers()
+    random.shuffle(servers)
+    return servers
+
+
+async def choose_non_coordinator(manager: ManagerClient, servers):
+    """ Return a server that is not the coordinator.
+    """
+    coord_id = await get_topology_coordinator(manager)
```

Comment:
BTW I looked into implementation of this and I see some problems with it (cc @tgrabiec)
- it actually returns Raft leader for the group, not the topology coordinator; the topology coordinator does not need to be the leader -- it follows soon after, but generally it does not start "atomically" with the leader
- it takes a `group_id` parameter so the implementation  appears to be prepared for the future -- when there will be multiple groups -- but it routes the request to shard 0. But not all groups will live on shard 0

But these are minor problems and fixing them is out of scope of this PR.

Still I kind of dislike the approach of "finding coordinator" in general, because the coordinator (or the raft leader) can in theory migrate at any time. The logic of "find coordinator -> do something to the coordinator (or not-coordinator)" assumes that the coordinator won't change before the "do something", which is in general wrong. 

In the case of our test, a more robust approach -- and I think it would result in actually much less code -- would be injecting the failure on all servers. Then no matter who the coordinator is, it will work.

But I don't reject this approach, you did the randomization to reproduce the problem often and verified it no longer reproduces with this approach, so if it works I can accept it.

Reply:
Yes, it looks like at least for the test the leader (/coordinator) don't migrate - even in the failing test in #18614 the `s[1]` was the leader since the very beginning, just that it was the second server in the list and not the first one.

Without the randomization I wasn't even able to reproduce the issue despite using `--repeat 100`, but with randomization it is reproduced easily (and after the fix the `--repeat 100` is all green again).

Injecting the failure on all servers would work too I suppose.